### DISCUSSION TITLE
no decode delegate on a possible messed up install

### DISCUSSION THREAD

**Discussion by skibumdon37** [ImageMagick/ImageMagick#5716] [Help]
I am having trouble with images from my canon 90d in CR3 format, or even converted to tiff.
My errors are basically this:

convert  b.tif test.jpg
convert: no decode delegate for this image format `TIF' @ error/constitute.c/ReadImage/741.
convert: no images defined `test.jpg' @ error/convert.c/ConvertImageCommand/3342.

exiftool says this:

 exiftool b.tif | grep -i tif
File Name                       : b.tif
File Type                       : TIFF
File Type Extension             : tif
MIME Type                       : image/tiff

My system is a windows 11, running wsl2, ubuntu 20.04.
I thought the issue would be that I needed to get imagemagick installed with libraw 0.20. 
I had libraw 0.19 installed with apt get-installed, so I attempted to remove it. However when I do a find on libraw in the /usr directory, I get 
 find . -name "libraw*" -print 2>/dev/null  (partial list)
./include/libraw
./include/libraw/libraw_alloc.h
./include/libraw/libraw_version.h
./include/libraw/libraw_internal.h
./include/libraw/libraw_datastream.h
./include/libraw/libraw_const.h
./include/libraw/libraw_types.h
./include/libraw/libraw.h
./lib/x86_64-linux-gnu/pkgconfig/libraw.pc
./lib/x86_64-linux-gnu/pkgconfig/libraw_r.pc
./lib/x86_64-linux-gnu/libraw.so.19
./lib/x86_64-linux-gnu/libraw_r.a
./lib/x86_64-linux-gnu/libraw_r.so.19.0.2
./lib/x86_64-linux-gnu/libraw.so
./lib/x86_64-linux-gnu/libraw_r.so
./lib/x86_64-linux-gnu/libraw_r.so.19
./lib/x86_64-linux-gnu/libraw.so.19.0.2
./lib/x86_64-linux-gnu/libraw.a
./share/doc/libraw19
./share/doc/libraw-dev
./local/include/libraw
./local/include/libraw/libraw_alloc.h
./local/include/libraw/libraw_version.h
./local/include/libraw/libraw_internal.h
./local/include/libraw/libraw_datastream.h
./local/include/libraw/libraw_const.h
./local/include/libraw/libraw_types.h
./local/include/libraw/libraw.h
./local/lib/pkgconfig/libraw.pc
./local/lib/pkgconfig/libraw_r.pc
./local/lib/libraw_r.a
./local/lib/libraw.so
./local/lib/libraw.so.20

So while I have libraw 0.20.0 installed (more on this below), I have some 0.19 files lurking. I do not know if this would screw up the libraw install or the imagemagick install.

I installed libraw as follows

tar xzvf LibRaw-0.20.0.tar.gz
autoreconf --install
./configure # with optional args
make
sudo make install

Then for imagemagick, I did 

sudo apt-get remove imagemagick
git clone https://github.com/ImageMagick/ImageMagick.git ImageMagick-7.1.0
./configure --with-modules 
make 
sudo make install 

I verified

convert --version
Version: ImageMagick 7.1.0-52 beta Q16-HDRI x86_64 aea87b538:20221016 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): freetype jng jpeg lcms ltdl png raw x xml zlib
Compiler: gcc (9.4)

I can make the logo.gif file, just am not handling stuff out of the canon (I used canon software to convert cr3 to tif as the test.

**Comment by snibgo** [comment]
As you have IM v7, I suggest you use "magick", not "convert".

Your command "convert b.tif test.jpg" fails, but it doesn't use libraw, so don't worry about that. Have you built with libtiff? If not, then that is the problem.

What is the complete text output from "magick -version"? Does it include tiff?

Try these:
```
magick rose: x.tiff

magick x.tiff x.jpg
```

**Reply by skibumdon37** [reply]
seems like it is not built in?

 magick -version
Version: ImageMagick 7.1.0-52 beta Q16-HDRI x86_64 aea87b538:20221016 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): freetype jng jpeg lcms ltdl png raw x xml zlib
Compiler: gcc (9.4)

However your two magick commands did work
My tiff files though still fail on the decode delegate

How do I build with libtiff?

I did try 

 magick djs_2022_10_21_2944.CR3 cr3.tif
magick: no encode delegate for this image format `CR3' @ error/constitute.c/WriteImage/1400.

but
magick djs_2022_10_21_2944.CR3 cr3.jpg
did work.

Having tif would be nice as deepskystacker saves as tif, but is not necessary as I bring the image into affinity photo ok

**Comment by fmw42** [comment]
you need to use the install tool on Windows and not try to install manually.

**Comment by snibgo** [comment]
> seems like it is not built in?
> Delegates (built-in): freetype jng jpeg lcms ltdl png raw x xml zlib

Your installation of IM does not have TIFF as a built-in delegate.

You have Windows. Why are you building IM from source? You can install a pre-built binary, which includes TIFF. See https://imagemagick.org/script/download.php#windows

**Reply by skibumdon37** [reply]
This is a linux build, albeit under windows wsl2/ubuntu

**Comment by fmw42** [comment]
Use Cygwin

**Comment by skibumdon37** [comment]
I did find another github post on the libtiff:

https://gist.github.com/nickferrando/fb0a44d707c8c3efd92dedd0f79d2911

and I did the libtiff install with apt-get, redid configure, make and install

tiffs work but generates warnings which I can ignore.

 magick -version
Version: ImageMagick 7.1.0-52 beta Q16-HDRI x86_64 aea87b538:20221016 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): freetype jbig jng jpeg lcms ltdl lzma png raw tiff x xml zlib
Compiler: gcc (9.4)

$magick ./light/Autosave.tif test.jpg
$magick: Unknown field with tag 50000 (0xc350) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/950.
$magick: Unknown field with tag 50002 (0xc352) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/950.
$magick: Unknown field with tag 50008 (0xc358) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/950.

Thanks for the help!!

**Comment by snibgo** [comment]
You can use "-quiet" to suppress warnings.
